src/components/RichTextEditor/Toolbar.tsx

```diff
@@ -32,32 +31,49 @@ import {
     MARK_UNDERLINE,
     usePlateEditorRef,
 } from "@udecode/plate";
-import React, { FC } from "react";
+import React, { FC, ReactElement } from "react";
 import { ELEMENT_CHECK_ITEM } from "./plugins/checkboxListPlugin";
 import { TextStyleDropdown } from "./TextStyleDropdown/TextStyleDropdown";
+import { defaultActions, EditorActions } from "./utils/actions";
 import { TextStyleType } from "./utils/getTextStyles";
 
 type ToolbarProps = {
     editorId?: string;
     textStyles?: TextStyleType[];
+    actions?: EditorActions[][];
 };
 
-type ButtonGroupProps = {
-    testId?: string;
-    children: JSX.Element | JSX.Element[];
-};
-
-export const Toolbar: FC<ToolbarProps> = ({ editorId, textStyles }) => {
+export const Toolbar: FC<ToolbarProps> = ({ editorId, textStyles, actions = [] }) => {
     const editor = usePlateEditorRef(editorId);
-
-    const ButtonGroup: FC<ButtonGroupProps> = ({ testId, children }) => (
-        <div
-            data-test-id={testId}
-            className="tw-flex tw-items-center tw-border-r last:tw-border-r-0 tw-px-3 tw-py-2 tw-border-black-5"
-        >
-            {children}
-        </div>
-    );
+    const toolbarActions = actions.length > 0 ? actions : defaultActions;
+    const toolbarComponents: Record<EditorActions, ReactElement> = {
```

Comment:
I'd move this outside of the component and pass in the editorId and textStyles:
```suggestion
    const toolbarComponents = (editorId?: string, textStyles?: TextStyleType[]): Record<EditorActions, ReactElement> => {
```

Reply:
Changed